core/vtk/ttkScalarFieldCriticalPoints/ttkScalarFieldCriticalPoints.cpp

```diff
@@ -20,7 +20,9 @@ using namespace ttk;
 vtkStandardNewMacro(ttkScalarFieldCriticalPoints);
 
 ttkScalarFieldCriticalPoints::ttkScalarFieldCriticalPoints() {
-
+#ifdef TTK_ENABLE_MPI
+  hasMPISupport_ = true;
```

Comment:
I'd recommend to move that to the constructor of the *base* class when available.
in this example `core/base/scalarFieldCriticalPoints/`